### DISCUSSION TITLE
How to use i18n in js file

### DISCUSSION THREAD

**Discussion by yuuid** [quasarframework/quasar#12592] [General - Components / Directives / etc] [answered]
The project currently has three languages. 
Switch through `$t ('')` in Vue file according to official documents.
However, in the JS file, if you use the official `quasar Lang. getlocale() ` method can only obtain the current locale of the browser. But what I want is the language environment in the current project.
Has anyone ever encountered this problem? How to solve it?

**Comment by AlaaDq** [comment, marked as answer]
you can export instance of your i18n boot file like this
```js
const i18n = setupI18n();
export default boot(async ({ app }) => {
  // Set i18n instance on app.failed
  app.use(i18n);
});

export { i18n };
```
and then import it in your script and access the current locale like below
```js
import { i18n} from "src/boot/i18n";
i18n.global.locale
```

other way you can access it from  global quasar helper i guess.
hope this will be helpfull

**Reply by mpont91** [reply]
This solution works. But if you change the language with a selector component it won't be reactive... Any idea to solve this?

**Comment by danb4r** [comment]
For current quasar version:

```
import { boot } from 'quasar/wrappers';

import { createI18n } from 'vue-i18n';

import messages from 'src/i18n';

let i18n = createI18n({});

export default boot(({ app }) => {
  i18n = createI18n({
    locale: 'en-US',
    messages,
  });

  // Set i18n instance on app
  app.use(i18n);
});

export { i18n };
```

And on the external script to use t('somethig')...

```
import { i18n } from 'src/boot/i18n';

const t = i18n.global.t;
```
